### PR TITLE
Added .travis.yml for Travis CI (PS-5.6)

### PR DESCRIPTION
**Pull request by inikep**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
+    -DENABLE_DOWNLOADS=1
+    -DWITHOUT_TOKUDB=ON
```
Comment (percona-ysorokin) [inline .travis.yml]: Both RocksDB and TokuDB must be enabled.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=Debug
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
+    -DENABLE_DOWNLOADS=1
+    -DWITHOUT_TOKUDB=ON
```
Comment (percona-ysorokin) [inline .travis.yml]: Both RocksDB and TokuDB must be enabled.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+  # Test "RelWithDebInfo" compilation
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
```
Comment (laurynas-biveinis) [inline .travis.yml]: -DWITH_LIBEVENT=system ?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  # Test "Debug" compilation
+  - rm -rf *
+  - CC=$CC_COMPILER CXX=$CXX_COMPILER cmake ..
+    -DCMAKE_BUILD_TYPE=Debug
+    -DMYSQL_MAINTAINER_MODE=ON
+    -DBUILD_CONFIG=mysql_release
+    -DFEATURE_SET=community
+    -DENABLE_DTRACE=OFF
+    -DWITH_SSL=$SSL
+    -DWITH_ZLIB=system
```
Comment (laurynas-biveinis) [inline .travis.yml]: Likewise

**Comment by inikep** [comment]
I added tests with and without `-DWITH_READLINE=system` and `-DWITH_PAM=1`. I also added `-DWITH_LIBEVENT=bundled` and `-DWITH_LIBEVENT=system`.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**